dataloaders/helpers.py

```diff
@@ -0,0 +1,141 @@
+    train_sampler = SubsetRandomSampler(train_indices)
+    test_sampler = SubsetRandomSampler(test_indices)
+
+    train_loader = DataLoader(dataset, batch_size=batch_size, sampler=train_sampler, drop_last=drop_last, pin_memory=True)
+    test_loader = DataLoader(dataset, batch_size=batch_size, sampler=test_sampler, drop_last=drop_last, pin_memory=True)
+
+    return train_loader, test_loader
+
+
+def prepare_dataset_with_elmo_tokenizer(dataset, text_columns: List[str], max_length: int):
+    """Tokenizes 'text' field and returns dataset with each column of `text_columns` transformed to token IDs,
+        as well as a 'label' column.
+    """
+    elmo_tokenizer = MosesTokenizer('en', no_escape=True)
+    def text_to_ids(e):
+        for col in text_columns:
+            text = elmo_tokenizer(e[col])
+            # TODO: Figure out how to do this successfully in batch.
+            text_ids = batch_to_ids([text])[0]
+            # Pad to max length.
+            text_ids = text_ids[:max_length]
+            pad_size = max_length - len(text_ids)
+            if pad_size > 0:
+                padding = torch.zeros(pad_size, 50, dtype=int)
+                text_ids = torch.cat((text_ids, padding), dim=0)
+            e[col] = text_ids
+        return e
+
+    # Caching sometimes causes weird issues with .map(), if you see that then
+    # add the load_from_cache_file=False argument below.
+    dataset = dataset.map(text_to_ids, batched=False, load_from_cache_file=False)
+    return dataset
+
+
+def dataloader_from_dataset(
+        dataset: datasets.Dataset,
+        text_columns: List[str], label_columns: List[str],
+        batch_size: int, shuffle: bool, drop_last: bool
+    ) -> DataLoader:
+    """Creates a dataloader from `dataset` with `text_ids` and `labels` keys. Makes sure things
+    are collated properly and returned as a batched (...text_columns, ...label_columns) tuple.
+    """
+    def collate_fn(batch):
+        """`batch` is a list with `batch_size` elements. Each element is a dict with
+        `label` and `text_ids` keys."""
+        d = collections.defaultdict(list)
+        for el in batch:
+            for k in text_columns:
+                d[k].append(torch.stack(el[k])) # shape (seq_length, 50)
+            for k in label_columns:
+                d[k].append(el[k]) # shape ()
+        # stack all lists into tensors and return in the proper order
+        return tuple(torch.stack(d[k]) for k in text_columns) + tuple(torch.stack(d[k]).float() for k in label_columns)
+    return DataLoader(dataset,
+        batch_size=batch_size, collate_fn=collate_fn,
+        shuffle=True, drop_last=drop_last, pin_memory=True
+    )
+
+
+def load_rotten_tomatoes(batch_size: int, max_length: int, drop_last: bool = True, num_examples: int = None) -> Tuple[DataLoader, DataLoader]:
+    """Returns tuple of (train_dataloader, test_dataloader)."""
+    dataset = datasets.load_dataset('rotten_tomatoes').shuffle(seed=42)
```

Comment:
Does it matter if `seed=42` or will it be overridden by `torch.manual_seed(args.random_seed)`?

Reply:
Ok fixed